### DISCUSSION TITLE
CSS / PostCSS styles order

### DISCUSSION THREAD

**Discussion by mattkrins** [mantinedev/mantine#4958] [Q&A] [answered]
Any idea why all my postcss modules are being applied before the mantine core/styles in head?
My import '@mantine/core/styles.css'; is in the base App.tsx and all other components are children of App.tsx, however the core styles are always imported/injected after all the other styles from each components postcss module:
![image](https://github.com/mantinedev/mantine/assets/2367602/daf918ed-d430-4701-9d4b-dcc15e4bde98)

Currently I need to add !important to every single rule to use css modules.

**Comment by rtivital** [comment]
Styles order is controlled by the framework that you are using. If you are not able to control styles, use CSS Layers – https://mantine.dev/styles/mantine-styles/#css-layers

**Reply by mattkrins** [reply]
I'm guessing this is caused by Vite then: https://github.com/vitejs/vite/issues/3924
NextJS and Vite seem to be the most popular frameworks used with Mantine, but [both](https://github.com/vercel/next.js/issues/16630) have this [import](https://github.com/vitejs/vite/issues/3924) issue?
What framework **does** Mantine support which doesn't have this issue?

**Reply by rtivital** [reply, marked as answer]
Mantine provides styles via `.css` files that's all. I cannot control how third-party tools inject styles. You can insert styles any way your project requires, the way provided in the documentation is a recommendation that I've tested to work with the template mentioned in the documentation. If you still want to import styles and not depend on the order of imports – use CSS layers.
